### PR TITLE
[Modules] Updated Cache/Redis to new dependency approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated Cache/Redis to new dependency approach

| Pipeline |
| - |
| [Cache: Redis](https://github.com/Azure/ResourceModules/actions/workflows/ms.cache.redis.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Cache/redis/.test/common/dependencies.bicep [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            }
+            registrationEnabled: false
+        }
+    }
+}
+
+@description('The resource ID of the created Virtual Network Subnet.')
+output subnetResourceId string = virtualNetwork.properties.subnets[0].id
+
+@description('The resource ID of the created Virtual Network Subnet.')
```
Comment (eriqua) [inline modules/Microsoft.Cache/redis/.test/common/dependencies.bicep:46]:
```suggestion
@description('The resource ID of the created Private DNS Zone.')
```

File: modules/Microsoft.Cache/redis/.test/common/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    name: '<<namePrefix>>${serviceShort}001'
+    capacity: 2
+    diagnosticLogCategoriesToEnable: [
+      'ApplicationGatewayAccessLog'
+      'ApplicationGatewayFirewallLog'
+    ]
+    diagnosticMetricsToEnable: [
+      'AllMetrics'
+    ]
+    diagnosticSettingsName: 'redisdiagnostics'
```
Comment (eriqua) [inline modules/Microsoft.Cache/redis/.test/common/deploy.test.bicep:52]: Should we align diagnostic to other tests, including the deployment of the shared diagnostics dependencies module?

Comment (MrMCake) [reply modules/Microsoft.Cache/redis/.test/common/deploy.test.bicep:52]: I'll try. Strange that it wasn't there.

Comment (MrMCake) [reply modules/Microsoft.Cache/redis/.test/common/deploy.test.bicep:64]: after a fix of the diagnostic log options - done

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**